woocommerce/woocommerce
# Add rich text editor component

### All Submissions:






### Changes proposed in this Pull Request:



This PR creates a rich text editor.  Note that this is not the final version, but an initial base for us to use.  A number of follow-ups are needed to make this production ready.

While Storybook examples have been created, the data registry behavior is quirky outside of a WP environment.  Wrapping the stories with a `RegistryProvider` allows them to initially work, but selected blocks aren't detected in the registry and we are unable to switch between block types.  If anyone has ideas as to why this might be, I'm all ears.

<img width="869" alt="Screen Shot 2022-09-12 at 1 39 42 PM" src="https://user-images.githubusercontent.com/10561050/189755821-a1bdf3ba-4567-490c-b324-a3353f9eaf8f.png">


### How to test the changes in this Pull Request:

1. Import and use the rich text editor inside a WCA page:
```
import { RichTextEditor } from '@woocommerce/components';

<RichTextEditor onChange={ () => console.log('changed') } />
```
2. Optionally, enqueue some GB assets for added styling:
```
		wp_enqueue_style( 'wp-edit-post' );
		wp_enqueue_style( 'wp-format-library' );
		wp_enqueue_editor();
		do_action( 'enqueue_block_editor_assets' );
```
3. Check the various blocks and formatting buttons in the toolbar, attempting to switch back and forth between block types

### Other information:




### FOR PR REVIEWER ONLY:
